var/spack/repos/builtin/packages/py-pandas/package.py

```diff
@@ -128,3 +128,9 @@ class PyPandas(PythonPackage):
     # https://pandas.pydata.org/pandas-docs/stable/getting_started/install.html#optional-dependencies
 
     skip_modules = ["pandas.tests", "pandas.plotting._matplotlib", "pandas.core._numba.kernels"]
+
+    def flag_handler(self, name, flags):
+        if name == "cflags":
+            if self.spec.satisfies("@0.24.2 %oneapi"):
```

Comment:
This is a pretty specific version, how did you even discover this? Does it really not affect any other versions? Do we really even need this version?

Reply:
It is the version the concretizer picked for me - I don’t know why. I don’t know whether it affects other versions and don’t have time to investigate right now. We are trying to get these last fixes in for our end-of-August E4S release. The newest OneAPI compiler turns on certain warnings as errors be default where prior releases may not have done this to the same extent.

Reply:
This is something I can follow up on after this release is finished at the end of August.

Reply:
Looking through recent CI logs I can see that at least for `py-pandas@1.5.3 %oneapi@2023.2.1` this is not an issue.